docs/DeveloperGuide.md

```diff
@@ -368,42 +412,174 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
   Use case ends.
 
-* 3a. The given index is invalid.
+* 3a. The given name is invalid. 
+  - 3a1. MediContact shows an error message.
+
+    Use case ends.
+
+* 3b. No specified field or any field specified is in the wrong format.
+  - 3b1. MediContact shows an error message.
+
+    Use case ends.
+
 
-    * 3a1. AddressBook shows an error message.
+**Use case: Filter a person**
 
-      Use case resumes at step 2.
+**MSS**
+
+1. User requests to filter contacts and specifies the criteria.
+2. Medication shows a list of persons matching details.
+
+   Use case ends.
+
+**Extensions**
+
+- 1a. Criterion is missing.
+    - 1a1. MediContact shows an error message.
+
+      Use case ends.
+
+- 1b. Wrong format in input
+    - 1b1. MediContact shows an error message specifying the incorrect format.
+  
+      Use case ends.
 
 **Use case: Find a person**
 
 **MSS**
 
 1. User requests to find contact
 2. User inputs the find command with contact details
-3. Medicontact shows a list of persons matching details
+3. MediContact shows a list of persons matching details
+
    Use case ends.
 
 **Extensions**
 
 - 2a. Necessary field is missing
-  - 2a1. Medicontact shows an error message indicating which field is missing
-  - 2a2. Use case ends.
+  - 2a1. MediContact shows an error message indicating which field is missing.  
+  
+    Use case ends.
+  
 - 2b. Wrong format in input
-  - 2b1.Medicontact shows an error message specifying the incorrect format.
-  - 2b2. Use case ends.
-- 2c. No match in users
-  - 2b1.Medicontact shows an error message specifying that there are no users that match the query.
-  - 2b2. Use case ends.
+  - 2b1.MediContact shows an error message specifying the incorrect format.
+  
+    Use case ends.
+  
+**Use case: Import address book**
```

Comment:
"Imports contact info from the given json file into MediContact." This is from our DG. Maybe can try update the use case closer to this?

Reply:
Since in UG we state as 'Exporting the data' and 'Saving the data' , do you think 'Import data' suitable here?